kratos/python_scripts/assign_vector_by_direction_process.py

```diff
@@ -69,13 +69,34 @@ def __init__(self, Model, settings ):
 
         self.model_part = Model[settings["model_part_name"].GetString()]
 
-        # Construct the component by component parameter objects
-        x_params = KratosMultiphysics.Parameters("{}")
-        y_params = KratosMultiphysics.Parameters("{}")
-        z_params = KratosMultiphysics.Parameters("{}")
+        # Get domain size
+        # Note that we check both the current model part and the root one
+        if self.model_part.ProcessInfo.Has(KratosMultiphysics.DOMAIN_SIZE):
+            domain_size = self.model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+            if domain_size not in [2,3]:
+                root_model_part = self.model_part.GetRootModelPart()
+                if root_model_part.ProcessInfo.Has(KratosMultiphysics.DOMAIN_SIZE):
+                    domain_size = root_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+                else:
+                    domain_size = 3
+                    warn_msg = f"DOMAIN_SIZE is found neither in '{self.model_part.Name}' nor in root model part '{self.model_part.GetRootModelPart().Name}' ProcessInfo containers. Defaulting to 3."
+                    KratosMultiphysics.Logger.PrintWarning("AssignVectorByDirectionProcess", warn_msg)
+        else:
+            root_model_part = self.model_part.GetRootModelPart()
+            if root_model_part.ProcessInfo.Has(KratosMultiphysics.DOMAIN_SIZE):
+                domain_size = root_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+            else:
+                domain_size = 3
+                warn_msg = f"DOMAIN_SIZE is found neither in '{self.model_part.Name}' nor in root model part '{self.model_part.GetRootModelPart().Name}' ProcessInfo containers. Defaulting to 3."
+                KratosMultiphysics.Logger.PrintWarning("AssignVectorByDirectionProcess", warn_msg)
+
+        # Check the obtained domain size value
+        if domain_size not in [2,3]:
+            raise ValueError(f"Domain size must be either 2 or 3. Found value {domain_size} in model part '{self.model_part.FullName()}'.")
```

Comment:
You could give a hint here that if domain size is 1 the scalar process should be used?

Reply:
Done!